src/librustdoc/html/templates/item_union.html

```diff
@@ -19,5 +19,5 @@ <h2 id="fields" class="fields small-section-header">
         {{ self.document_field(field) | safe }}
     {% endfor %}
 {% endif %}
-{{ self.render_assoc_items() | safe }}
-{{ self.document_type_layout() | safe }}
+{{ self::item_template_render_assoc_items(self.borrow()) | safe }}
```

Comment:
I just thought about something: could we instead of calling `.borrow()` change the API of `item_template_render_assoc_items` to take a `&RefCell<T> where T: ItemTemplate` or something equivalent to not have to write `.borrow()` every time? I'm not sure if it's possible though. If you could confirm it, it'd be awesome!

Reply:
`self::item_template_render_attributes_in_pre(self) | safe` expands to 
- Without `.borrow()`: `&::askama::filters::safe(::askama::Html, self::item_template_render_attributes_in_pre(&(self)))?`
- With `.borrow()`: `&::askama::filters::safe(::askama::Html, self::item_template_render_attributes_in_pre({self.borrow()}))?`

I'm not sure why the function didn't catch `&impl ItemTemplate` here for the argument without `.borrow()`...

Reply:
One thing that I found out to remove `.borrow()` calls is by:
- Implementing `ItemTemplate` trait for `&ItemUnion` (additional `impl` block)
- Remove the trait bounds for implementors of `ItemTemplate`, i.e. remove `askama::Template` and `fmt::Display` bounds, because the derive macro doesn't implement for `&T`

I have to admit that this might look "hacky", I'll try to explore other options for a while and see how it goes

Reply:
Can't replicate the error behaviour, I have no idea on this

```rust
use std::fmt;

// askama's
trait Template {}

// our new trait
trait ItemTemplate: Template + fmt::Display {}

// concrete
struct ItemUnion;
impl ItemTemplate for ItemUnion {}
impl Template for ItemUnion {}
impl fmt::Display for ItemUnion {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "ItemUnion")
    }
}

// fn taking item template
fn item_template_document(v: &impl ItemTemplate) {
    println!("{}", v);
}

fn main() {
    let item = ItemUnion;
    // no compilation error below, why?
    item_template_document(&(item));
}
```

Reply:
I'm pretty much stuck at this point, what do you think? @GuillaumeGomez

Reply:
I really like the first suggestion which also seems to be the simplest to be implemented (but maybe I'm wrong there). Thanks for taking a look! So what do you prefer: making the changes here directly or opening an issue and fixing it in a follow-up PR?

Reply:
Oops, I laid them out wrongly, they should be implemented together.

Doing only the 1st item will cause errors, and the compiler suggested to do the 2nd item to fix it.

Reply:
Oh I see. Well, that still seems doable. Then same question for how you prefer things to roll out. :)

Reply:
> I really like the first suggestion which also seems to be the simplest to be implemented (but maybe I'm wrong there). Thanks for taking a look! So what do you prefer: making the changes here directly or opening an issue and fixing it in a follow-up PR?

Nevertheless, I'd prefer to do a follow-up PR and open up a new issue. I would like to see opinions from others (in case there exists something simpler).

Reply:
Let's go this way then! I r+ this PR. Please open an issue. ;)